web/elm/src/Routes.elm

```diff
@@ -89,159 +143,238 @@ buildRoute : Concourse.Build -> String
 buildRoute build =
     case build.job of
         Just j ->
-            Build j.teamName j.pipelineName j.jobName build.name |> toString
+            Build j.teamName j.pipelineName j.jobName build.name HighlightNothing
+                |> toString
 
         Nothing ->
-            OneOffBuild (Basics.toString build.id) |> toString
+            OneOffBuild (Basics.toString build.id) HighlightNothing
+                |> toString
 
 
 jobRoute : Concourse.Job -> String
 jobRoute j =
-    Job j.teamName j.pipelineName j.name |> toString
+    Job j.teamName j.pipelineName j.name Nothing |> toString
 
 
 pipelineRoute : { a | name : String, teamName : String } -> String
 pipelineRoute p =
-    Pipeline p.teamName p.name |> toString
+    Pipeline p.teamName p.name [] |> toString
 
 
 dashboardRoute : String
 dashboardRoute =
-    Dashboard |> toString
+    Dashboard Nothing |> toString
 
 
 dashboardHdRoute : String
 dashboardHdRoute =
     DashboardHd |> toString
 
 
+showHighlight : Highlight -> String
```

Comment:
...Why is this now in routes? wouldn't it make more sense to keep it in Steptree, the only place it's used?

Reply:
it's the inverse of `parseHighlight`, and my hope is that when we upgrade to 0.19 we will use https://package.elm-lang.org/packages/elm/url/latest/Url-Builder to construct links. It could still go where you're suggesting, but I guess it feels kinda nice for `Routes` to become the interface to the `Url` package in the future.

Reply:
Ahhh, I see. Cool.